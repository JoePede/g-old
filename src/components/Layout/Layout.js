/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { connect } from 'react-redux';
import normalizeCss from 'normalize.css';
import { createSSESub } from '../../actions/sseSubs';

// external-global styles must be imported in your JS.
import s from './Layout.css';
import Header from '../Header';
import Footer from '../Footer';

class Layout extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    createSSESub: PropTypes.func.isRequired,
  };
  componentDidMount() {
    this.props.createSSESub();
  }
  render() {
    return (
      <div>
        <Header />
        <div className={s.content}>
          {this.props.children}
        </div>
        <Footer />
      </div>
    );
  }
}

const mapDispatch = {
  createSSESub,
};
export default connect(null, mapDispatch)(withStyles(normalizeCss, s)(Layout));
