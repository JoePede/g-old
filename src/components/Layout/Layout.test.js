/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { IntlProvider } from 'react-intl';
import App from '../App';
import Layout from './Layout';

const middlewares = [thunk.withExtraArgument({ fetch: () => {} })];
const mockStore = configureStore(middlewares);
const initialState = {
  runtime: {
    initialNow: 1501773564066,
    availableLocales: ['de-DE'],
  },
  intl: {
    initialNow: 1501773564065,
    locale: 'de-DE',
    newLocale: null,
    messages: {
      'de-DE': {},
    },
  },
  entities: {
    users: { byId: {} },
  },
  ui: {
    activityCounter: {
      feed: 0,
      proposals: 0,
    },
  },
};

describe('Layout', () => {
  it('renders children correctly', () => {
    const store = mockStore(initialState);
    const initialNow = new Date();
    const wrapper = renderer
      .create(
        <IntlProvider locale="en" initialNow={initialNow}>
          <App
            context={{
              intl: {
                initialNow: 1501773564065,
                locale: 'de-DE',
                newLocale: null,
                messages: {
                  'de-DE': {},
                },
              },
              fetch: () => {},
              insertCss: () => {},
              store,
            }}
            store={store}
          >
            <Layout>
              <div className="child" />
            </Layout>
          </App>
        </IntlProvider>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
