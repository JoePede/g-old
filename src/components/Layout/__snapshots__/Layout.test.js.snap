// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <div
        className="root"
        role="navigation"
      >
        <div
          className="navBar"
        >
          <a
            className="link"
            href="/feed"
            onClick={[Function]}
          >
            Feed
          </a>
          <a
            className="link"
            href="/proposals/active"
            onClick={[Function]}
          >
            <span>
              Proposals
            </span>
          </a>
          <a
            className="link"
            href="/surveys"
            onClick={[Function]}
          >
            <span>
              Surveys
            </span>
          </a>
          <a
            className="link"
            href="/admin"
            onClick={[Function]}
          >
            <span>
              Admin
            </span>
          </a>
          <a
            className="link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
        </div>
        <div
          className="menu"
        >
          <div
            className="root column"
            onClick={null}
          >
            <button
              className="button plain"
              disabled={false}
              onClick={[Function]}
            >
              <span
                className="icon"
              >
                 
                <svg
                  height="20px"
                  style={
                    Object {
                      "display": "inline-block",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 32 32"
                  width="20px"
                >
                  <path
                    d="M2 6h28v6h-28zM2 14h28v6h-28zM2 22h28v6h-28z"
                    style={
                      Object {
                        "fill": "grey",
                      }
                    }
                    transform=""
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="banner"
      >
        <h1
          className="bannerTitle"
        >
          <span>
            G O L D
          </span>
        </h1>
        Government by OnLine Democracy
      </div>
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="child"
    />
  </div>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        © G O L D
      </span>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <div>
        <span>
          <span>
            Deutsch
          </span>
           
        </span>
      </div>
    </div>
  </div>
</div>
`;
